docs/content/development/modules.md

```diff
@@ -8,7 +8,7 @@ Druid uses a module system that allows for the addition of extensions at runtime
 
 ## Writing your own extensions
 
-Druid's extensions leverage Guice in order to add things at runtime.  Basically, Guice is a framework for Dependency Injection, but we use it to hold the expected object graph of the Druid process.  Extensions can make any changes they want/need to the object graph via adding Guice bindings.  While the extensions actually give you the capability to change almost anything however you want, in general, we expect people to want to extend one of a few things.
+Druid's extensions leverage Guice in order to add things at runtime.  Basically, Guice is a framework for Dependency Injection, but we use it to hold the expected object graph of the Druid process.  Extensions can make any changes they want/need to the object graph via adding Guice bindings.  While the extensions actually give you the capability to change almost anything however you want, in general, we expect people to want to extend one of the things listed below.  This means that we honor semantic versioning for changes that affect the interfaces called out on this page, but other interfaces are deemed "internal" anad can be changed in an incompatible manner even between patch releases.
```

Comment:
I'm referring to this on http://semver.org/

> Major version zero (0.y.z) is for initial development. Anything may change at any time. The public API should not be considered stable.

We're offering something stronger than that.